### PR TITLE
feat: add tool params

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR adds tool params.

### CODE REVIEW

File: packages/valory/skills/task_execution/behaviours.py [hunk 1]
```diff
@@ -396,7 +399,8 @@ def _prepare_task(self, task_data: Dict[str, Any]) -> None:
         executing_task = cast(Dict[str, Any], self._executing_task)
         executing_task["timeout_deadline"] = time.time() + self.params.task_deadline
         executing_task["tool"] = task_data["tool"]
-        executing_task["model"] = task_data.get("model", None)
+        executing_task["model"] = task_data.get("model", tool_params.pop('default_model', None))
```
Comment (0xArdi) [inline packages/valory/skills/task_execution/behaviours.py:402]: default model will be picked up here

File: packages/napthaai/customs/prediction_request_rag_claude/component.yaml [hunk 2]
```diff
@@ -11,6 +11,8 @@ fingerprint:
 fingerprint_ignore_patterns: []
 entry_point: prediction_request_rag_claude.py
 callable: run
+params:
+  default_model: claude-3-haiku-20240307
```
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_rag_claude/component.yaml:15]: any param defined here will be reflected in the metadata stored on-chain

File: packages/napthaai/customs/prediction_request_rag/component.yaml [hunk 3]
```diff
@@ -7,10 +7,12 @@ license: Apache-2.0
 aea_version: '>=1.0.0, <2.0.0'
 fingerprint:
   __init__.py: bafybeibt7f7crtwvmkg7spy3jhscmlqltvyblzp32g6gj44v7tlo5lycuq
-  prediction_request_rag.py: bafybeie4rafsx7m7mfkdgd2n2ovfm3gwk5ybjr5cz6s7r3oz545ag2wwuy
+  prediction_request_rag.py: bafybeibtvuddvbhjlyd4sbk7rwz4mcsr4hiigfgrpdhzwa6vn6bhb6fboy
 fingerprint_ignore_patterns: []
 entry_point: prediction_request_rag.py
 callable: run
+params:
+  default_model: claude-3-sonnet-20240229
```
Comment (cyberosa) [inline packages/napthaai/customs/prediction_request_rag/component.yaml:15]: Where is this parameter used? At the python file the default model is hard coded

File: packages/napthaai/customs/prediction_request_reasoning/component.yaml [hunk 4]
```diff
@@ -7,10 +7,12 @@ license: Apache-2.0
 aea_version: '>=1.0.0, <2.0.0'
 fingerprint:
   __init__.py: bafybeib36ew6vbztldut5xayk5553rylrq7yv4cpqyhwc5ktvd4cx67vwu
-  prediction_request_reasoning.py: bafybeibyncgeeyrlcqhdbsdzzama7aah44jzedqo3zcplvuc45goykwjli
+  prediction_request_reasoning.py: bafybeidiabgnlc453spgrdn7rhhl2xc3aa6zqeukkw2bthndbugtjf6bya
 fingerprint_ignore_patterns: []
 entry_point: prediction_request_reasoning.py
 callable: run
+params:
+  default_model: claude-3-sonnet-20240229
```
Comment (cyberosa) [inline packages/napthaai/customs/prediction_request_reasoning/component.yaml:15]: same question here

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py [hunk 5]
```diff
@@ -299,11 +297,11 @@ def count_tokens(text: str, model: str) -> int:
...
 def multi_queries(
     prompt: str,
-    engine: str,
+    model: str,
     num_queries: int,
     counter_callback: Optional[Callable[[int, int, str], None]] = None,
-    temperature: Optional[float] = LLM_SETTINGS[DEFAULT_MODEL]["temperature"],
-    max_tokens: Optional[int] = LLM_SETTINGS[DEFAULT_MODEL]["default_max_tokens"],
+    temperature: Optional[float] = LLM_SETTINGS["gpt-4-0125-preview"]["temperature"],
+    max_tokens: Optional[int] = LLM_SETTINGS["gpt-4-0125-preview"]["default_max_tokens"],
```
Comment (cyberosa) [inline packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:304]: you don't use the same default model as defined in the component.yaml file

File: packages/napthaai/customs/prediction_request_rag/component.yaml (hunk 3 above)
Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_rag/component.yaml:15]: You can see in Ardian's comments above.

**Review by cyberosa — APPROVED** [review]

File: packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py (hunk 5 above)
Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_rag/prediction_request_rag.py:304]: I don't think this default can ever be used (was the same before I made the changes). Could change them from optional in future I guess.

**Pull request merged**